packages/relay/src/lib/eth.ts

```diff
@@ -447,6 +447,41 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * Gets a transaction by the provided hash
+   *
+   * @param hash
+   */
+  async getTransactionByHash(hash: string) {
+    this.logger.trace('getTransactionByHash(hash=%s)', hash);
+    const contractResult = await this.mirrorNodeClient.getContractResult(hash);
+    if (contractResult === null || contractResult.hash === undefined) {
+      return null;
+    }
+
+    return new Transaction({
+      accessList: contractResult.access_list,
+      blockHash: contractResult.block_hash.substring(0, 66),
+      blockNumber: contractResult.block_number,
+      chainId: contractResult.chain_id,
+      from: contractResult.from.substring(0, 42),
+      gas: contractResult.gas_used,
+      gasPrice: contractResult.gas_price,
+      hash: contractResult.hash.substring(0, 66),
+      input: contractResult.function_parameters,
+      maxPriorityFeePerGas: contractResult.max_priority_fee_per_gas,
+      maxFeePerGas: contractResult.max_fee_per_gas,
+      nonce: contractResult.nonce,
+      r: contractResult.r.substring(0, 66),
+      s: contractResult.s.substring(0, 66),
+      to: contractResult.to.substring(0, 42),
```

Comment:
Same question as from:, is this just defensive?

Reply:
This was my suggestion and just defensive. Actually checked MirrorNode response and min size is 40 and max is 42.
I can fix this in my following PR's